### ISSUE TITLE
[camera][android] Unit tests fail

### ISSUE THREAD

**Issue by svenjacobs** [flutter/flutter#71709]
I'm integrating a Flutter module into an Android app in an "add 2 app" situation. After adding `camera` (0.5.8+11) to the Flutter module, the Android unit tests fail. When running `./gradlew test` in my Android project I get the following error:

```
> Task :camera:compileDebugUnitTestJavaWithJavac FAILED
/Users/sven/.pub-cache/hosted/pub.dartlang.org/camera-0.5.8+11/android/src/test/java/io/flutter/plugins/camera/DartMessengerTest.java:103: warning: [unchecked] unchecked cast
        return (Map<String, String>) StandardMethodCodec.INSTANCE.decodeEnvelope(sentMessage);
                                                                                ^
  required: Map<String,String>
  found:    Object
error: warnings found and -Werror specified
1 error
1 warning
```

I'm wondering how this failing unit test was not discovered yet because the unchecked cast is ignored in the test's [source code](https://github.com/flutter/plugins/blob/9fef1c7f289caa81d683cdb5c7a5fb006fc840e5/packages/camera/camera/android/src/test/java/io/flutter/plugins/camera/DartMessengerTest.java#L98-L99) but obviously [warnings are treated as errors](https://github.com/flutter/plugins/blob/9fef1c7f289caa81d683cdb5c7a5fb006fc840e5/packages/camera/camera/android/build.gradle#L3).

Flutter version 1.22.4.

**Comment by TahaTesser** [comment]
Hi @svenjacobs 
Can you please provide your `flutter doctor -v` and a complete reproducible minimal code sample in a GitHub repository.
Thank you

**Comment by svenjacobs** [comment]
Output of `flutter doctor -v`:

```
[✓] Flutter (Channel stable, 1.22.4, on Mac OS X 10.15.7 19H15 darwin-x64, locale en-DE)
    • Flutter version 1.22.4 at /Users/sven/.fvm/versions/stable
    • Framework revision 1aafb3a8b9 (3 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/sven/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /Users/sven/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin installed
    • Dart plugin version 203.5981.152

[✓] VS Code (version 1.51.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • Pixel (mobile) • FA69L0316725 • android-arm64 • Android 10 (API 29)

! Doctor found issues in 1 category.
```
(by the way, the warning is wrong. Flutter & Dart plugins are installed in Android Studio)

The example project can be found [here](https://github.com/svenjacobs/flutter-camera-issue-71709). Please run `./gradlew test` in `android` folder.

**Comment by TahaTesser** [comment]
Hi @svenjacobs 
I can reproduce the issue, thanks for filing the issue

<details> 
<summary>logs</summary> 

```bash
PS C:\Users\Taha> cd C:\Users\Taha\AndroidStudioProjects\flutter-camera-issue-71709\flutter\.android
PS C:\Users\Taha\AndroidStudioProjects\flutter-camera-issue-71709\flutter\.android> ./gradlew test

Welcome to Gradle 5.6.2!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.2/release-notes.html


> Task :camera:compileDebugUnitTestJavaWithJavac FAILED
C:\Users\Taha\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\camera-0.5.8+17\android\src\test\java\io\flutter\plugins\camera\DartMessengerTest.java:99: warning: [unchecked] unchecked cast
    return (Map<String, String>) StandardMethodCodec.INSTANCE.decodeEnvelope(sentMessage);
                                                                            ^
  required: Map<String,String>
  found:    Object
error: warnings found and -Werror specified
1 error
1 warning

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':camera:compileDebugUnitTestJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 10s
115 actionable tasks: 85 executed, 30 up-to-date
PS C:\Users\Taha\AndroidStudioProjects\flutter-camera-issue-71709\flutter\.android>
```

</details>

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.19042.630], locale en-US)
    • Flutter version 1.22.4 at C:\Code\flutter_stable
    • Framework revision 1aafb3a8b9 (3 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.51.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[✓] Connected device (1 available)
    • sdk gphone x86 (mobile) • emulator-5556 • android-x86 • Android 11 (API 30) (emulator)

! Doctor found issues in 1 category.
```

</details>

**Comment by mvanbeusekom** [comment]
Fixed with PR flutter/plugins#3316.
